File: Framework-Samples/NHibernate/VisualBasic/Sample.Data/ManagerObjects/InventoryManager.vb
Imports System
Imports System.Collections.Generic
Imports System.Text

Imports NHibernate
Imports Sample.Data.Generated.BusinessObjects
Imports Sample.Data.Generated.Base

Namespace Sample.Data.Generated.ManagerObjects
	'Public Partial Interface IInventoryManager
	'	Inherits IManagerBase(Of Inventory, System.String)
	'End Interface

	Partial Class InventoryManager
		Inherits ManagerBase(Of Inventory, System.String)
		Implements IInventoryManager
	End Class
End Namespace

File: Framework-Samples/NHibernate/CSharp/Sample.Data/ManagerObjects/InventoryManager.cs
using System;
using System.Collections.Generic;
using System.Text;

using NHibernate;
using Sample.Data.Generated.BusinessObjects;
using Sample.Data.Generated.Base;

namespace Sample.Data.Generated.ManagerObjects
{
    public partial interface IInventoryManager : IManagerBase<Inventory, string>
    {
	}
	
	partial class InventoryManager : ManagerBase<Inventory, string>, IInventoryManager
    {
	}
}